src/main/scala/com/advancedtelematic/director/http/AssignmentsResource.scala

```diff
@@ -51,24 +55,33 @@ class AssignmentsResource(extractNamespace: Directive1[Namespace])
             val f = createAssignments(ns, req).map(_ => StatusCodes.Created)
             complete(f)
           }
+        } ~
+        patch {
+          entity(as[CancelAssignments]) { cancelAssignments =>
+            val a = deviceAssignments.cancel(ns, cancelAssignments.cancelAssignments).flatMap { canceledAssignments =>
+              Future.traverse(canceledAssignments) { canceledAssignment =>
```

Comment:
This should somehow go in a single transaction. Probably means that this logic should go inside a class that does this and creates a composed DBIO containing all changes

Reply:
Didn't know that!

Reply:
I was reading this wrong, my mistake.

To clean up this resource, would it make sense to pass the msgbuspublisher to `deviceAssignments.cancel` and that method would push messages to bus?